api/eventsingress_api.md

```diff
@@ -0,0 +1,49 @@
+<!--:nav_order:2-->
+
+# Events Publishing API
+
+## Prerequisites
+
+* Create a project and a workspace in the [`Adobe Developer Console`](https://www.adobe.io/apis/experienceplatform/console/docs.html#!AdobeDocs/adobeio-console/master/projects-empty.md)
+* Add the `I/O Management API` in this `Adobe Developer Console` workspace 
+* [Generate a JWT token](https://www.adobe.io/apis/experienceplatform/console/docs.html#!AdobeDocs/adobeio-console/master/credentials.md)
+* Using [Adobe I/O Events Provider API](provider_api.md) 
+   * create your own `Custom Events Provider`
+   * create at least one `Event Metadata` associated with the above
+
+## Test Drive
+
+Once its `Event Metadata` is persisted in Adobe I/O Events (see above prerequisites),
+your `Custom Events Provider` can start publishig its 
+[CloudEvents]( https://cloudevents.io) to Adobe I/O Events publishing endpoint.
+
+Please follow [CloudEvents v1.0 specification](https://github.com/cloudevents/spec/blob/v1.0/spec.md), 
+here is a sample `curl` command:
+
+     curl  --location --request POST  \
+             --url https://eventsingress.adobe.io \
+             --header "x-api-key: $api_key" \
+             --header "Authorization: Bearer $jwt_token" \
+             --header 'Content-Type: application/cloudevents+json' \
+             --data '{
+               "datacontenttype": "application/json",
+               "specversion": "1.0",
+               "source": "urn:uuid:'"${provider_id}"'",
+               "type": "'"${event_code}"'",
+               "id": "'"${your_event_id}"'",
+               "data": "your event json payload"
+             }'
+
+
+The environment variables used in this `curl` command are computed from the above prerequisites
+* `api_key` is the api-key associated with your `Adobe Developer Console` workspace
+* `jwt_token` is a jwt token generated using the set up from the same workspace
+* `provider_id` is your `Custom Events Provider` uuid generated by [Adobe I/O Events Provider API](../api/provider_api.md)
+* `event_code` is the `Custom Events Provider`'s `Event Metadata`'s `event_code` as persisted using [Adobe I/O Events Provider API](../api/provider_api.md)
+* `your_event_id` is any id of your choice, 
+ensure that `source + id` is unique for each distinct event see [CloudEvents spec](https://github.com/cloudevents/spec/blob/v1.0/spec.md#id)
+*  as for the value of `data` in the CloudEvents body payload, it can be any json payload.
+
+
+
+        
```

Comment:
_Nitpick:_ Can you please remove the extra new lines in the end?

Reply:
done